examples/route_guide/util/span_interception.go

```diff
@@ -0,0 +1,78 @@
+package util 
+
+import (
+	"time"
+	"fmt"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"github.com/opentracing/opentracing-go"
+	"github.com/lightstep/lightstep-tracer-go"
+)
+
+type metadataReaderWriter struct {
+	metadata.MD
+}
+
+func (w metadataReaderWriter) Set(key, val string) {
+	fmt.Println(key, " -> ", val)
+	w.MD[key] = append(w.MD[key], val)
+}
+
+func (w metadataReaderWriter) ForeachKey(handler func(key, val string) error) error {
+	for k, vals := range w.MD {
+		for _, v := range vals {
+			if dk, dv, err := metadata.DecodeKeyValue(k, v); err == nil {
+				if err = handler(dk, dv); err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func InitSpan(tracer opentracing.Tracer, ctx context.Context, operation string) (opentracing.Span, context.Context){
+	span := tracer.StartSpanWithOptions(opentracing.StartSpanOptions{
+		OperationName: operation,
+	    StartTime: time.Now(),
+	})
+
+	span.LogEvent(operation+"_called")
+	md := metadata.New(make(map[string]string))
+	//inject span rep into context 
+	tracer.Inject(span,opentracing.TextMap,metadataReaderWriter{md})
+
+	return span, metadata.NewContext(context.Background(), md)
```

Comment:
sorry, whoops... use `ctx` here for sure. and make it the first param to `InitSpan`